### PR TITLE
Use resource_path for datatypes linter

### PR DESCRIPTION
**Pull request by bernt-matthias**
https://github.com/galaxyproject/galaxy/pull/17600#discussion_r1854230715

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by jdavcs** [comment]
Unless this is a bug fix (I guess not?), it needs to target the dev branch.

File: lib/galaxy/tool_util/linters/datatypes.py [hunk 1]
```diff
@@ -5,20 +5,22 @@
...
 if TYPE_CHECKING:
     from galaxy.tool_util.lint import LintContext
     from galaxy.tool_util.parser import ToolSource
 
-DATATYPES_CONF = os.path.join(os.path.dirname(__file__), "datatypes_conf.xml.sample")
+DATATYPES_CONF = resource_path(linters, "datatypes_conf.xml.sample")
 
 
-def _parse_datatypes(datatype_conf_path: str) -> Set[str]:
+def _parse_datatypes(datatype_conf_path) -> Set[str]:
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/linters/datatypes.py:23]: @bernt-matthias Unfortunately dropping the type annotation here hides issues when `datatype_conf_path` is processed by `galaxy.util.parse_xml()` . Do you mind if I open a separate PR that addresses them? I went down a rabbit hole for the last couple of days unfortunately :D

Comment (bernt-matthias) [reply lib/galaxy/tool_util/linters/datatypes.py:23]: > Do you mind if I open a separate PR that addresses them?

No at all. Please do so. 

> I went down a rabbit hole for the last couple of days unfortunately

Good that you found out again.

**Comment by bernt-matthias** [comment]
Superseded by https://github.com/galaxyproject/galaxy/pull/19331
